app/src/main/java/com/arctouch/codechallenge/detail/DetailViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package com.arctouch.codechallenge.detail
+
+import android.databinding.ObservableField
+import com.arctouch.codechallenge.base.BaseViewModel
+import com.arctouch.codechallenge.extensions.formatGenres
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.usecase.MovieDetailUseCase
+import com.arctouch.codechallenge.util.MovieImageUrlBuilder
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import java.text.SimpleDateFormat
+import javax.inject.Inject
+
+class DetailViewModel @Inject constructor(private val movieImageUrlBuilder: MovieImageUrlBuilder, private val movieDetailUseCase: MovieDetailUseCase) : BaseViewModel() {
+
+    val backdrop = ObservableField<String>()
+    val poster = ObservableField<String>()
+    val title = ObservableField<String>()
+    val releaseDate = ObservableField<String>()
+    val overview = ObservableField<String>()
+    val genres = ObservableField<String>()
+
+    fun init(movieId: Int) {
+
+        disposable = movieDetailUseCase
+                .getMovieDetails(movieId)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe {
+                    prepareData(it)
+                }
+    }
+
+    private fun prepareData(movie: Movie) {
+        title.set(movie.title)
+        overview.set(movie.overview)
+//        releaseDate.set(formatDate(movie.releaseDate))
+        releaseDate.set(movie.releaseDate)
+        genres.set(movie.formatGenres())
+        backdrop.set(movie.backdropPath?.let { movieImageUrlBuilder.buildBackdropUrl(it) })
+        poster.set(movie.posterPath?.let { movieImageUrlBuilder.buildPosterUrl(it) })
+    }
+
+    private fun formatDate(date: String?): String {
+        return date?.let {
```

Comment:
With Kotlin we can achieve great power with his conciseness. But sometimes, it could make the code slightly hard to read, mainly for starters in your team or that doesn't have too much background in Kotlin. A more readable version for this code could be:

    private fun formatDate(unformattedDate: String?): String {
        return unformattedDate?.asDate() ?: ""
    }

    private fun String.asDate(): String {
        val formatBackend = SimpleDateFormat("yyyy/MM/dd")
        val sdf = SimpleDateFormat("dd/MM/yyyy")
        return sdf.format(formatBackend.parse(this))
    }